teams_rules.go

```diff
@@ -77,6 +79,12 @@ type UntrustedCertSettings struct {
 	Action TeamsGatewayUntrustedCertAction `json:"action"`
 }
 
+type NotificationSettings struct {
```

Comment:
until we have per product namespacing, we'll need to prefix this to avoid polluting the global scope.

```suggestion
type TeamsNotificationSettings struct {
```